Helm chart generates duplicate ConfigMap value

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Enabling the `eni` integration and explicitly enabling `endpointRoutes` in the chart should be redundant, as `eni.enabled` will enable endpoint routing, but it looks like similar to #13999 the `enable-endpoint-routes` key in the resultant `ConfigMap` is not deduplicated.

Minimal reproduction:
```console
bash-5.2$ helm template cilium cilium/cilium --version 1.15.2 \
  --namespace cilium \
  --set endpointRoutes.enabled=true \
  --set eni.enabled=true > cilium-eni.yaml
bash-5.2$ yamllint -d relaxed --no-warnings cilium-eni.yaml
cilium-eni.yaml
  191:3     error    duplication of key "enable-endpoint-routes" in mapping  (key-duplicates)
  793:1     error    trailing spaces  (trailing-spaces)
  809:1     error    trailing spaces  (trailing-spaces)
```

### Cilium Version

1.15.2

### Kernel Version

N/A

### Kubernetes Version

N/A

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/install/kubernetes/cilium/templates/cilium-configmap.yaml
+++ b/install/kubernetes/cilium/templates/cilium-configmap.yaml
@@ -500,7 +500,9 @@ data:
 {{- end }}
 
 {{- if .Values.eni.enabled }}
+  {{- if not .Values.endpointRoutes.enabled }}
   enable-endpoint-routes: "true"
+  {{- end }}
   auto-create-cilium-node-resource: "true"
 {{- if .Values.eni.updateEC2AdapterLimitViaAPI }}
   update-ec2-adapter-limit-via-api: "true"

